### ISSUE TITLE
bad interaction between graphicx.sty and luaotfload

### ISSUE THREAD

**Issue by emojifreak** [latex3/luaotfload#124]
latex-base-dev version: 
cat-version: 2020-02-01 pre-release 2
cat-date:    2019-11-20 19:32:31 +0100

luaotfload version:
cat-version: 3.11
cat-date:    2019-11-11 10:37:48 +0100

When I compile the following by `lualatex-dev`

```latex
\documentclass{article}
\usepackage{graphicx}
\font\fontB={[./TwemojiMozilla.ttf]:+colr;+ccmp;+dist;}
\begin{document}
% example-image-1x1.pdf from texlive/2019/texmf-dist/tex/latex/mwe/example-image-1x1.pdf
%\includegraphics[width=20pt]{example-image-1x1.pdf}% (*)
{\fontB ^^^^2603}!{\fontB ^^^^^^01f986}
\end{document}
```

I got 
![bxtest9](https://user-images.githubusercontent.com/53028623/70310578-55429300-1853-11ea-9d2d-00c83676a699.png)

This is fine and expected 👌

But when I compile the following by `lualatex-dev`

```latex
\documentclass{article}
\usepackage{graphicx}
\font\fontB={[./TwemojiMozilla.ttf]:mode=harf;+colr;+ccmp;+dist;}
\begin{document}
% example-image-1x1.pdf from texlive/2019/texmf-dist/tex/latex/mwe/example-image-1x1.pdf
\includegraphics[width=20pt]{example-image-1x1.pdf}% (*)
{\fontB ^^^^2603}!{\fontB ^^^^^^01f986}
\end{document}
```

I got 
![bxtest7](https://user-images.githubusercontent.com/53028623/70310658-87ec8b80-1853-11ea-9726-570cccf42097.png)

and when I compile

```latex
\documentclass{article}
\usepackage{graphicx}
\font\fontB={[./TwemojiMozilla.ttf]:+colr;+ccmp;+dist;}
\begin{document}
% example-image-1x1.pdf from texlive/2019/texmf-dist/tex/latex/mwe/example-image-1x1.pdf
\includegraphics[width=20pt]{example-image-1x1.pdf}% (*)
{\fontB ^^^^2603}!{\fontB ^^^^^^01f986}
\end{document}
```

I got 
![bxtest8](https://user-images.githubusercontent.com/53028623/70310701-9f2b7900-1853-11ea-891a-18af9fbcf3e8.png)

Tha last two PDFs are unexpected 😭

The TwemojiMozilla.ttf is from https://github.com/mozilla/twemoji-colr/releases
The same strange behavior can be observed with Segoe UI Emoji of MS Windows. So it does not seem a font problem. Since the strange behavior happens with and without `mode=harf`, it does not seem a problem in the underlying ConTeXt.

The above example was given by @zr-tex8r at https://github.com/zr-tex8r/BXcoloremoji/issues/3#issuecomment-562430370

**Comment by zauguin** [comment]
Thanks for the report.

> Since the strange behavior happens with and without `mode=harf`, it does not seem a problem in the underlying ConTeXt.

Usually I would agree, but in this case the problem independently exists in harf and ConTeXt code. For `harf` fonts it is fixed in the dev branch now, for the ConTeXt fontloader I'll have to report it to their team.

**Comment by emojifreak** [comment]
@u-fischer  @zauguin The symptom remains with `\font\fontB={[./TwemojiMozilla.ttf]:+colr;+ccmp;+dist;}`
and `\font\fontB={[./TwemojiMozilla.ttf]:mode=harf;+colr;+ccmp;+dist;}`
produces another wrong PDF as
![Screenshot at 2020-02-15 12-44-22](https://user-images.githubusercontent.com/53028623/74581283-f6d32700-4ff0-11ea-8d1d-1aed961b523b.png)

luaotfload is 3.12 and lualatex-dev is `This is LuaHBTeX, Version 1.11.2 (TeX Live 2020/dev) `
Could you have another look at this issue and reopen this?

**Comment by zauguin** [comment]
@emojifreak The problem without `mode=harf` is "expected" (I didn't send this to the ConTeXt guys yet because I am unable to reproduce it in ConTeXt for some reason) but the `mode=harf` version works fine here.

Your harf error looks like an independent issue though. Maybe you could create a new issue where you also upload the full log file for the `mode=harf` case?

**Comment by emojifreak** [comment]
@zauguin 
> but the `mode=harf` version works fine here.

I am sorry. I mistyped `mode=harf` as `+harf`. `mode=harf` produced an expected PDF by the latest versions of luaotfload and lualatex-dev. Sorry again for wasting your time.

**Comment by emojifreak** [comment]
I verified that this issue still exists in luaotfload 3.13.

**Comment by u-fischer** [comment]
> The problem without `mode=harf` is "expected" (I didn't send this to the ConTeXt guys yet because I am unable to reproduce it in ConTeXt for some reason) but the `mode=harf` version works fine here.

@zauguin I managed to create a context example (but I'm not completly sure if I'm using the newest context, so better test). tinypicture is a small 3x3 picture, but any one should do.  I used seguiemj.ttf as I have it on windows, but probbly the other emoji show the same problem. 

The problem appears as soon as a matrix is set. Then only the next char gets this too, but the other are displaced (they are in the pdf).


~~~~
\starttext
\catcode`\^= 7
\font\fontB={file:seguiemj.ttf:mode=node;+colr;+ccmp;+dist;}

 \saveimageresource  {tinypicture.pdf}
 \pdfextension save\relax
 \pdfextension setmatrix{1 0 0 1}%
 \useimageresource\lastsavedimageresourceindex
 \pdfextension restore\relax
 \fontB ^^^^^^01f986\fontB ^^^^2603 x

\stoptext
~~~~

**Comment by u-fischer** [comment]
actually a simple save/restore pair before the emoji font is already enough:

~~~~
\font\fontB={file:seguiemj.ttf:mode=node;+colr;+ccmp;+dist;}
\starttext
\catcode`\^= 7
\pdfextension save\relax
\pdfextension restore\relax

\fontB^^^^2603 ^^^^^^01f986 x
\stoptext
~~~~

**Comment by u-fischer** [comment]
@emojifreak it should be resolved in version 3.14 that I uploaded last night.

**Comment by emojifreak** [comment]
Yes, it is fixed in 3.14! Thanks!
